Cloudinary.Test/CloudinaryTest.cs

```diff
@@ -1094,6 +1094,43 @@ public void TestRename()
             Assert.AreEqual("ico", getResult.Format);
         }
 
+        [Test]
+        public void TestRenameNewType()
+        {
+            string publicId = string.Concat(m_suffix, "_renameNewType");
+            string newPublicId = string.Concat(m_suffix, "_renameNewType1");
+
+            var uploadParams = new ImageUploadParams()
+            {
+                PublicId = publicId,
+                File = new FileDescription(m_testImagePath),
+                Tags = m_test_tag,
+                Type = "upload"
+            };
+
+            var uploadResult = m_cloudinary.Upload(uploadParams);
+
+            RenameParams renameParams = new RenameParams(publicId, newPublicId)

```

Comment:
Why do we test here also plain rename?

Reply:
Additional verification, just in case. According to reliability theory - the more times a check is duplicated, the more reliable the whole system is. In this case, it checks how the method will behave under the new parameter - "NewType" and if the plain rename will break.

We can remove it if we really have to.

Reply:
Since the test doesn't check the entire system (at least it shouldn't), it should test the bare minimum functionality. redundant assertions increase the test execution time + might create non relevant noise.